server/src/test/java/com/objectcomputing/checkins/services/reports/ReportDataControllerTest.java

```diff
@@ -0,0 +1,133 @@
+package com.objectcomputing.checkins.services.reports;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.objectcomputing.checkins.services.TestContainersSuite;
+import com.objectcomputing.checkins.services.fixture.RoleFixture;
+import com.objectcomputing.checkins.services.fixture.MemberProfileFixture;
+import com.objectcomputing.checkins.services.memberprofile.MemberProfile;
+
+import io.micronaut.core.type.Argument;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.MutableHttpRequest;
+import io.micronaut.http.client.HttpClient;
+import io.micronaut.http.client.annotation.Client;
+import io.micronaut.http.client.exceptions.HttpClientResponseException;
+import io.micronaut.http.client.multipart.MultipartBody;
+
+import jakarta.inject.Inject;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static com.objectcomputing.checkins.services.role.RoleType.Constants.ADMIN_ROLE;
+import static com.objectcomputing.checkins.services.role.RoleType.Constants.MEMBER_ROLE;
+import static io.micronaut.http.MediaType.MULTIPART_FORM_DATA;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class ReportDataControllerTest extends TestContainersSuite implements MemberProfileFixture, RoleFixture {
+
+  @Inject
+  @Client("/services/report/data")
+  HttpClient client;
+
+  private MemberProfile regular;
+  private MemberProfile admin;
+  private final String basePath = "src/test/java/com/objectcomputing/checkins/services/reports/";
+
+  @BeforeEach
+  void createRolesAndPermissions() {
+    createAndAssignRoles();
+    regular = createADefaultMemberProfile();
+    admin = createAThirdDefaultMemberProfile();
+    assignAdminRole(admin);
+  }
+
+  @Test
+  void uploadReportData() {
+    final HttpRequest<?> request = postData(admin, ADMIN_ROLE);
+    final String response = client.toBlocking().retrieve(request);
+    assertNotNull(response);
+  }
+
+  @Test
+  void uploadReportDataWithoutPermission() {
+    final HttpRequest<?> request = postData(regular, MEMBER_ROLE);
+    HttpClientResponseException responseException =
+      assertThrows(HttpClientResponseException.class,
+                   () -> client.toBlocking().retrieve(request));
+  }
+
+  @Test
+  void getReportData() throws JsonProcessingException {
```

Comment:
We don't have any ReportKudos or ReportHours populated in the test. Should we add some so we can be confident it's working as expected?

Reply:
Will do!

Reply:
A test for `selfReviews`, `reviews` and `feedback` that checks the returned values are as expected (not just non-null) would be good too, as the `private List<Feedback> getFeedbackType(FeedbackType type)` has some complex logic in it, and it would be good to be able to refactor it with confidence at a later date 😄